plugins/redfish/fu-redfish-multipart-device.c

```diff
@@ -109,7 +109,14 @@ fu_redfish_multipart_device_write_firmware(FuDevice *device,
 	part = curl_mime_addpart(mime);
 	curl_mime_name(part, "UpdateFile");
 	(void)curl_mime_type(part, "application/octet-stream");
-	(void)curl_mime_filename(part, "firmware.bin");
+
+	if (fu_redfish_backend_get_vendor(backend) != NULL &&
+	    g_strcmp0(fu_redfish_backend_get_vendor(backend), "Dell") == 0) {
+		/* Dell's iDRAC requires a specific filename to know in advance how to flash it */
+		(void)curl_mime_filename(part, "firmware.exe");
```

Comment:
This feels like a hack; give me a few minutes and I'll see if I can give you some better API to use.

Reply:
Yup, will work on integrating that on Monday!